ui/src/app/cohort-search/modal/modal.component.tsx

```diff
@@ -0,0 +1,597 @@
+    marginBottom: '0.5rem',
+    padding: '0rem 1rem',
+    display: 'flex',
+    justifyContent: 'space-between',
+    alignItems: 'flex-end',
+    height: '2.5rem',
+    marginTop: '0.5rem',
+  }
+});
+
+const css = `
+  .modal-container {
+    height: 90vh;
+    width: 80vw;
+  }
+  .modal-container.demographics {
+    width: 50vw;
+    height: auto;
+  }
+  @media (max-width: 1500px) {
+  .modal-container {
+    width: 90vw;
+  }
+  .modal-container.demographics {
+    width: 70vw;
+  }
+  @media (max-width: 1200px) {
+    .modal-container, .modal-container.demographics {
+      width: 100vw;
+    }
+  }
+  @media (max-height: 700px) {
+    .modal-container {
+      height: 100vh;
+    }
+  }
+  @media (max-height: 400px) {
+    .modal-container.demographics {
+      height: 100vh;
+    }
+    .modal-container.demographics.age {
+      height: auto;
+    }
+  }
+  @media (max-height: 300px) {
+    .modal-container.demographics.age {
+      height: 100vh;
+    }
+  }
+}
+`;
+
+function initGroup(role: string, item: any) {
+  return {
+    id: generateId(role),
+    items: [item],
+    count: null,
+    temporal: false,
+    mention: TemporalMention.ANYMENTION,
+    time: TemporalTime.DURINGSAMEENCOUNTERAS,
+    timeValue: '',
+    timeFrame: '',
+    isRequesting: false,
+    status: 'active'
+  };
+}
+
+interface Selection extends Criteria {
+  parameterId: string;
+}
+
+interface Props {
+  closeSearch: () => void;
+  searchContext: any;
+  setSearchContext: (context: any) => void;
+}
+
+interface State {
+  attributesNode: Criteria;
+  autocompleteSelection: Criteria;
+  backMode: string;
+  conceptType: string;
+  count: number;
+  disableFinish: boolean;
+  groupSelections: Array<number>;
+  hierarchyNode: Criteria;
+  loadingSubtree: boolean;
+  mode: string;
+  open: boolean;
+  selectedIds: Array<string>;
+  selections: Array<Selection>;
+  title: string;
+  treeSearchTerms: string;
+}
+
+export class CBModal extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      attributesNode: undefined,
+      autocompleteSelection: undefined,
+      backMode: undefined,
+      conceptType: null,
+      count: 0,
+      disableFinish: false,
+      groupSelections: [],
+      hierarchyNode: undefined,
+      loadingSubtree: false,
+      mode: 'list',
+      open: false,
+      selectedIds: [],
+      selections: [],
+      title: '',
+      treeSearchTerms: '',
+    };
+  }
+
+  componentDidMount(): void {
+    const {searchContext: {domain, item, standard, type}} = this.props;
+    if (!this.state.open) {
+      const selections = item.searchParameters;
+      const selectedIds = selections.map(s => s.parameterId);
+      if (type === CriteriaType.DECEASED) {
+        this.selectDeceased();
+      } else {
+        const title = domain === DomainType.PERSON ? typeToTitle(type) : domainToTitle(domain);
+        let backMode, hierarchyNode, mode;
```

Comment:
nit: backMode and mode can be initialized to 'list', we will not need else then

Reply:
Good call, done.